src/api/responseFactory.js

```diff
@@ -0,0 +1,50 @@
+import { isInteger } from 'util/numeric';
+
+let payload = {};
+let metaData = {};
+
+const buildList = (mockResponse, page) => {
+  const totalItems = mockResponse.length;
+  const lastPage = page.pageSize === 0 ? 1 : Math.ceil(totalItems / page.pageSize);
+  const computedPage = page.page <= lastPage ? page.page : lastPage;
+
+  payload = page.pageSize === 0 ?
+    mockResponse :
+    mockResponse.splice((computedPage * page.pageSize) - page.pageSize, page.pageSize);
+
+  metaData = {
+    page: computedPage,
+    pageSize: page.pageSize,
+    totalItems,
+    lastPage,
+  };
+};
+
+const buildPayload = (mockResponse, page) => {
+  if (mockResponse instanceof Array) {
```

Comment:
`Array.isArray()` is usually recommended, see [MDN](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/isArray)